README
NAME
    IRC::Toolkit - Useful IRC objects and utilities

SYNOPSIS
      ## Import the most commonly used Tookit:: modules
      ##  (Case, Colors, CTCP, ISupport, Masks, Modes, Parser)
      use IRC::Toolkit;

      ## Import a list of modules:
      use IRC::Toolkit qw/
        CTCP
        Masks
        Modes
        Numerics
      /;

      ## ... or individually:
      use IRC::Toolkit::Numerics;

DESCRIPTION
    A collection of useful IRC-related utilities. See their respective
    documentation, below.

    Modules that export functions use Exporter::Tiny, which is quite
    flexible; see the Exporter::Tiny docs for details.

    IRC::Message::Object; Objects representing incoming or outgoing IRC
    events

    IRC::Mode::Single; Objects representing a single mode change

    IRC::Mode::Set; Objects representing a set of mode changes

    IRC::Toolkit::Case; RFC-compliant case folding tools

    IRC::Toolkit::Colors; Color/format code interpolation & removal

    IRC::Toolkit::CTCP; CTCP quoting and extraction tools

    IRC::Toolkit::ISupport; ISUPPORT (numeric 005) parser

    IRC::Toolkit::Masks; Hostmask parsing and matching tools

    IRC::Toolkit::Modes; Mode-line parsing tools

    IRC::Toolkit::Numerics; IRC numerics translation to/from RPL or ERR
    names

    IRC::Toolkit::Parser; Functional interface to POE::Filter::IRCv3

    IRC::Toolkit::TS6; Produce sequential TS6 IDs

    IRC::Toolkit::Role::CaseMap; A Role for classes that track IRC
    casemapping settings

SEE ALSO
    IRC::Utils

    Parse::IRC

AUTHOR
    Jon Portnoy <avenj@cobaltirc.org>

    Portions of code are derived from IRC::Utils, Net::IRC, and
    POE::Filter::IRC::Compat, which are copyright their respective authors;
    these items are documented where they are found.

    Licensed under the same terms as Perl.



Changes
Revision history for Perl module IRC::Toolkit

0.089001 2015-01-03

  - IRC::Message::Object;
    add TO_JSON method
    drop unnecessary ->has_command predicate

0.088002 2014-08-16

  - Fix $isupport->can()

  - Happy CPAN Day!

0.088001 2014-01-19

  - IRC::Toolkit::Modes;
    array_to_mode supports IRC::Mode::Set objects
    mode_to_array accepts IRC::Toolkit::ISupport ->chanmodes objects

  - IRC::Toolkit::Colors;
    Add has_color / strip_color

  - IRC::Toolkit::Numerics;
    Add charybdis QUIETLIST numerics

  - IRC::Mode::Set;
    Remove deprecated new_from_* methods (went away in 0.08)

0.087000 2013-10-26

  - IRC::Toolkit::Case;
    Add |rfc1459| infix comparison operator
    Add overloaded IRC::Toolkit::Case::MappedString objects
    (and an 'irc_str()' function to construct them)

0.086000 2013-10-23

  - Use List::Objects::WithUtils list objects for IRC::Message::Object
    attributes

0.085000 2013-09-12

  - Switch to Exporter::Tiny

  - Warn on unknown casemaps in lc/uc/eq

  - Minor cleanups & optimizations

0.084002 2013-08-11

  - perl-5.8.9 compatible release

0.084001 2013-08-10

  - Minor cleanups & optimizations

  - Fix 'Changes' to be in line with CPAN::Changes::Spec

0.084000 2013-03-15

  - ISupport array-type retvals are now List::Objects::WithUtils::Arrays

  - Prefer ratbox label for numeric '010'

0.083000 2013-03-14

  - Add OO interface to IRC::Toolkit::Numerics

0.082000 2013-03-14

  - Add IRC::Toolkit::Numerics

  - Documentation fixes

0.081001 2013-03-03

  - Missing dependency on namespace::clean

0.081000 2013-03-01

  - Add IRC::Toolkit::Role::CaseMap

  - Add ISupport->extban->as_string & ->chanmodes->as_string

  - Import parse_isupport() by default on 'use IRC::Toolkit'

0.080000 2013-02-23

  - Incompatible change:
    IRC::Mode::Set->new_from_params renamed ->clone_from_params
    IRC::Mode::Set->new_from_mode   renamed ->clone_from_mode

  - Add IRC::Mode::Set->clone() and ->export()

  - IRC::Message::Object;
    Fix filter + raw_line combination at construction-time

0.075000 2013-02-23

  - Add IRC::Message::Object->truncate
  
  - Expose ->filter and ->colonify attribs

0.074000 2013-02-22

  - Add IRC::Toolkit::Modes::array_to_mode()

  - Document ISUPPORT EXTBAN parser

0.073000 2013-02-22

  - Add ELIST, EXTBAN, and TARGMAX support to ISupport parser

0.072000 2013-02-22

  - Add IRC::Toolkit::ISupport

0.071001 2013-02-21

  - Fix 'use IRC::Toolkit qw/ ... /' imports

  - Documentation fixes

0.071000 2013-02-10

  - Add IRC::Toolkit::TS6 ID generator

0.070003 2013-02-09

  - Missing dependency on namespace::clean

0.070002 2013-02-09

  - Documentation updates

0.070001 2013-02-09

  - Add IRC::Mode::Single, IRC::Mode::Set objects
    (extracted from IRC::Server::Pluggable)

0.06 2013-02-01

  - Fix bad test (wrt 5.17 hash ordering changes)

  - Explicitly require Test::More 0.88

0.05 2013-01-28

  - IRC::Toolkit::Modes; Kill use of '...'

0.04 2013-01-28

  - IRC::Message::Object; kill attribute writers
  
  - IRC::Toolkit::Parser;
    irc_line_from_ref also accepts IRC::Message::Object instances

0.03 2013-01-27

  - POD tweaks
  
  - Move dist to dzil

0.02 2013-01-27

  - POD/test tweaks

  - Fix repo metadata

0.01 2013-01-27

  - Initial release


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;



use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Useful IRC objects and utilities",
  "AUTHOR" => "Jon Portnoy <avenj\@cobaltirc.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "IRC-Toolkit",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "IRC::Toolkit",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Exporter::Tiny" => 0,
    "List::Objects::Types" => 1,
    "List::Objects::WithUtils" => 2,
    "Moo" => 1,
    "MooX::late" => "0.014",
    "POE::Filter::IRCv3" => "0.045",
    "Role::Tiny" => 0,
    "Scalar::Util" => 0,
    "Storable" => 0,
    "Sub::Infix" => 0,
    "Types::Standard" => 0,
    "namespace::clean" => 0,
    "overload" => 0,
    "strictures" => 1
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "0.089001",
  "test" => {
    "TESTS" => "t/*.t t/02_util/*.t t/05_role/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Exporter::Tiny" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Objects::Types" => 1,
  "List::Objects::WithUtils" => 2,
  "Moo" => 1,
  "MooX::late" => "0.014",
  "POE::Filter::IRCv3" => "0.045",
  "Role::Tiny" => 0,
  "Scalar::Util" => 0,
  "Storable" => 0,
  "Sub::Infix" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.88",
  "Types::Standard" => 0,
  "namespace::clean" => 0,
  "overload" => 0,
  "strictures" => 1
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



